src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2252,58 +2271,56 @@ private static bool ContainsModifier(SyntaxTokenList modifiers, SyntaxKind modif
                 AddAccessorIfAvailable(indexerMembers, indexer.SetMethod, diagnostics, checkName: true);
             }
 
-            var membersByName = MergeIndexersAndNonIndexers(membersAndInitializers.NonTypeNonIndexerMembers, indexerMembers);
+            var merged = mergeIndexersAndNonIndexers(membersAndInitializers.NonTypeNonIndexerMembers, indexerMembers);
+
+            var membersByName = merged.ToDictionary(s => s.Name, StringOrdinalComparer.Instance);
+            merged.Free();
             indexerMembers.Free();
 
             // Merge types into the member dictionary
             AddNestedTypesToDictionary(membersByName, GetTypeMembersDictionary());
 
             return membersByName;
-        }
 
-        /// <summary>
-        /// Merge (already ordered) non-type, non-indexer members with (already ordered) indexer members.
-        /// </summary>
-        private static Dictionary<string, ImmutableArray<Symbol>> MergeIndexersAndNonIndexers(ImmutableArray<Symbol> nonIndexerMembers, ArrayBuilder<Symbol> indexerMembers)
-        {
-            int nonIndexerCount = nonIndexerMembers.Length;
-            int indexerCount = indexerMembers.Count;
+            // Merge (already ordered) non-type, non-indexer members with (already ordered) indexer members.
+            static ArrayBuilder<Symbol> mergeIndexersAndNonIndexers(ImmutableArray<Symbol> nonIndexerMembers, ArrayBuilder<Symbol> indexerMembers)
```

Comment:
>ArrayBuilder<Symbol> [](start = 19, length = 20)

Why return an `ArrayBuilder<Symbol>` instead of `Dictionary<string, ImmutableArray<Symbol>>` as the original method did? (Do we need to make any changes to the original method?)